Mlem/Views/Shared/Composer/ResponseEditorView.swift

```diff
@@ -23,100 +23,145 @@ struct ResponseEditorView: View {
 
     private var isReadyToReply: Bool {
         editorBody.trimmed.isNotEmpty
     }
-
+    
     var body: some View {
-        ScrollView {
-            VStack(spacing: AppConstants.postAndCommentSpacing) {
-                // Post Text
-                TextField(
-                    "What do you want to say?",
-                    text: $editorBody,
-                    axis: .vertical
-                )
-                .lineLimit(AppConstants.textFieldVariableLineLimit)
-                .accessibilityLabel("Response Body")
-                .padding(AppConstants.postAndCommentSpacing)
-                .focused($focusedField, equals: .editorBody)
-                .onAppear {
-                    focusedField = .editorBody
+        NavigationStack(path: .constant(.init())) {
+            ScrollView {
+                VStack(spacing: AppConstants.standardSpacing) {
+                    // Post Text
+                    BodyEditorView(
+                        text: $editorBody,
+                        prompt: "What do you want to say?",
+                        bodyEditorModel: bodyEditorModel,
+                        attachmentModel: attachmentModel
+                    )
+                    .linkAttachmentModel(model: attachmentModel)
+                    .lineLimit(AppConstants.textFieldVariableLineLimit)
+                    .accessibilityLabel("Response Body")
+                    .padding(AppConstants.standardSpacing)
+                    .focused($focusedField, equals: .editorBody)
+                    .onAppear {
+                        focusedField = .editorBody
+                    }
+                    .onChange(of: editorBody) { newValue in
+                        if editorModel.showSlurWarning {
+                            slurMatch = siteInformation.instance?.firstSlurFilterMatch(newValue)
+                        }
+                    }
+                    
+                    Divider()
+                    
+                    infoView
                 }
-                .onChange(of: editorBody) { newValue in
-                    if editorModel.showSlurWarning {
-                        slurMatch = siteInformation.instance?.firstSlurFilterMatch(newValue)
+                .animation(.default, value: attachmentModel.imageModel?.state)
+                .animation(.default, value: slurMatch)
+                .padding(.bottom, AppConstants.editorOverscroll)
+            }
+            .scrollDismissesKeyboard(.automatic)
+            .overlay {
+                // Loading Indicator
+                if isSubmitting {
+                    ProgressView()
+                        .frame(maxWidth: .infinity, maxHeight: .infinity)
+                        .background(Color.gray.opacity(0.3))
+                        .accessibilityElement(children: .ignore)
+                        .accessibilityLabel("Submitting Response")
+                        .edgesIgnoringSafeArea(.all)
+                        .allowsHitTesting(false)
```

Comment:
We use this basic construct in a couple different places--maybe we want it in a reusable component?

Reply:
Done 👍